packages/react-native-reanimated/src/createAnimatedComponent/createAnimatedComponent.tsx

```diff
@@ -127,12 +129,21 @@ export function createAnimatedComponent(
     static displayName: string;
     static contextType = SkipEnteringContext;
     context!: React.ContextType<typeof SkipEnteringContext>;
+    nativeID = id++;
 
     constructor(props: AnimatedComponentProps<InitialComponentProps>) {
       super(props);
       if (isJest()) {
         this.jestAnimatedStyle = { value: {} };
       }
+      const entering = this.props.entering;
```

Comment:
General question: since we agree that `createAnimatedComponent` file is pretty heavy and unreadable, what would we think about making a specific manager for maintaining and handling  JS-side code for Layout Animations? (You can see [`NativeEventsManager`](https://github.com/software-mansion/react-native-reanimated/blob/main/packages/react-native-reanimated/src/createAnimatedComponent/NativeEventsManager.ts) for reference) I know it might bring a lot of re-testing and complicate review. (cc @piaskowyk @tjzel )

Reply:
Update: I've just noticed `animationsManager.ts`, but the idea is: is it possible to move all the calculations and specific logic to that file and keep only its API calls in the component file?

Reply:
I think this could be done in a separate PR to keep this one from being even more inflated.